### ISSUE TITLE
Change event is fired after Selectize element is destroyed

### ISSUE THREAD

**Issue by billsaysthis** [selectize/selectize.js#734]
One of our uses of Selectize requires that we destroy the (Backbone) view which contains the Selectize element when the user clicks one of the options. While doing this we're getting an error from MicroEvent.trigger on line 691:

```
      for (var i = 0; i < this._events[event].length; i++){
```

When called for the change event after already being called for the destroy event (we are calling $selectize_elem.destroy() before destroying the containing view), this._events[event].length causes a "Cannot read property 'length' of undefined" error.

If I change the offending line to the following, all's well:

```
      for (var i = 0; this._events[event] && i < this._events[event].length; i++){
```

Additionally in my view's destroy event I had to add a line (this.selectize is the Selectize element):

```
      this.selectize.destroy();
      this.selectize._events = {};
```

Without the second line, line 691 above triggers a JS error: "Cannot read property 'change' of undefined".

**Comment by GeorgeDewar** [comment]
I have encountered this issue also. In my case, upon deleting the last item from the Selectize control, the control is removed and destroy() is called, causing the problem.

I found a workaround based on yours that solved it without modifying the library code (though it's fragile). There were two listeners on the 'change' event, so the following made it happy (_selectize_ being my Selectize element):

`selectize._events = {change: []};`
